pkg/node/client.go

```diff
@@ -0,0 +1,165 @@
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (n *NodeClient) getBAR(ctx context.Context, barName, barNs string) (*v1alpha1.BucketAccessRequest, error) {
+	klog.Infof("getting bucketAccessRequest %q", fmt.Sprintf("%s/%s", barNs, barName))
+	bar, err := n.cosiClient.BucketAccessRequests(barNs).Get(ctx, barName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucketAccessRequest", fmt.Sprintf("%s/%s", barNs, barName), err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if bar == nil || !bar.Status.AccessGranted {
+	//	return nil, logErr(fmt.Errorf("BucketAccessRequest is nil or BucketAccessRequest does not grant access %q", fmt.Sprintf("%s/%s", barNs, barName)))
+	//}
+	if len(bar.Spec.BucketRequestName) == 0 {
+		return nil, logErr(fmt.Errorf("bucketAccessRequest.Spec.BucketRequestName unset"))
+	}
+	return bar, nil
+}
+
+func (n *NodeClient) getBA(ctx context.Context, baName string) (*v1alpha1.BucketAccess, error) {
+	klog.Infof("getting bucketAccess %q", fmt.Sprintf("%s", baName))
+	ba, err := n.cosiClient.BucketAccesses().Get(ctx, baName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucketAccess", fmt.Sprintf("%s", baName), err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if ba == nil || !ba.Status.AccessGranted {
+	//	return nil, logErr(fmt.Errorf("BucketAccess is nil or BucketAccess does not grant access %q", fmt.Sprintf("%s", baName)))
+	//}
+	return ba, nil
+}
+
+func (n *NodeClient) getBR(ctx context.Context, brName, brNs string) (*v1alpha1.BucketRequest, error) {
+	klog.Infof("getting bucketRequest %q", brName)
+	br, err := n.cosiClient.BucketRequests(brNs).Get(ctx, brName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucketRequest", fmt.Sprintf("%s/%s", brNs, brName), err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if br == nil || !br.Status.BucketAvailable {
+	//	return nil, logErr(fmt.Errorf("BucketRequest is nil or BucketRequest is not available yet %q", fmt.Sprintf("%s/%s", brNs, brName)))
+	//}
+	return br, nil
+}
+
+func (n *NodeClient) getB(ctx context.Context, bName string) (*v1alpha1.Bucket, error) {
+	klog.Infof("getting bucket %q", bName)
+	// is BucketInstanceName the correct field, or should it be BucketClass
+	bkt, err := n.cosiClient.Buckets().Get(ctx, bName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucket", bName, err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if bkt == nil || !bkt.Status.BucketAvailable {
+	//	return nil, logErr(fmt.Errorf("Bucket is nil or Bucket is not available yet %q", fmt.Sprintf("%s", bName)))
+	//}
+	return bkt, nil
+}
+
+func (n *NodeClient) GetResources(ctx context.Context, barName, barNs string) (bkt *v1alpha1.Bucket, secret *v1.Secret, err error) {
+	bar, err := n.getBAR(ctx, barName, barNs)
+	if err != nil {
```

Comment:
nit: slight consolidation by combining the assignment and the `if`

Reply:
Done!